views/StudyDetails.react.js

```diff
@@ -0,0 +1,197 @@
+import React, { useState, useEffect } from 'react'
+import { compose } from 'redux';
+import { withStyles } from '@material-ui/core/styles';
+import { connect } from 'react-redux';
+import Typography from '@material-ui/core/Typography';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Snackbar from '@material-ui/core/Snackbar';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+import Add from '@material-ui/icons/Add';
+import Delete from '@material-ui/icons/Delete';
+import Header from './components/Header';
+import Sidebar from './components/Sidebar';
+import getCounts from './fe-utils/countUtil';
+import { fetchSubjects, fetchStudyDetails, deleteDetails } from './fe-utils/fetchUtil';
+import getDefaultStyles from './fe-utils/styleUtil';
+import { studyDetailStyles } from './styles/study_details';
+import { routes } from './routes/routes'
+import getAvatar from './fe-utils/avatarUtil';
+
+const StudyDetails = (props) => {
+  const { user, classes } = props
+
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [isLoading, setLoading] = useState(true)
+  const [sideBarState, setSideBarState] = useState({ totalDays: 0, totalStudies: 0, totalSubjects: 0 })
+  const [snackBarState, setSnackBarState] = useState({ errorOpen: false, message: '', autoHideDuration: 4000, })
+  const [studyDetailsList, setStudyDetailsList] = useState([]);
+  const [avatar, setAvatar] = useState('');
+
+  useEffect(() => {
+    fetchSubjects().then(acl => {
+      setSideBarState(getCounts({ acl }))
+    })
+    fetchStudyDetails().then(({ data }) => {
+      setStudyDetailsList(data)
+      setLoading(false)
+    })
+    setAvatar(getAvatar({ user: user }))
+
+  }, [])
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+  const handleCloseError = (event, reason) => {
+    if (reason === 'clickaway')return;
+    setSnackBarState(state => ({errorOpen: false, ...state }))
+  }
+  const handleChangeFile = async (e) => {
+    const file = e.target.files ? e.target.files[0] : '';
+    try {
+      const upload = await new Response(file).json()
+      const result = await window.fetch(`${routes.basePath}/api/v1/study-details/upload`, {
```

Comment:
should this be moved to a separate function like `fetchStudyDetails` or `deleteDetails`? Maybe `createStudyDetails`?